### PR TITLE
[v4] Fix some  wysiwyg issues

### PR DESCRIPTION
**Pull request by soupette**
### What does it do?

It fixes:

- [x] Selecting multiple lines and trying to set either the unordered and ordered lists, only the first entry is prefixed by it's list option

- [x] Selecting a line and trying to set ordered list will always give you 1. and not continue the ordered list (only way to do it is to press enter after the first entry)

- [x]  Underline is using the HTML code instead of standard markdown (there's no fix for this I think @derrickmehaffy)

- [x] Underline, Strikethrough, and code block all have an odd red color to them, like the syntax isn't correct

- [x] Pressing enter after a link gives some really odd index level on the next line

- [~]  Maximized window has it's scroll bars offset in a really weird way and doesn't get very big
  - [x] The scroll bar is fixed
  - [ ] The max height is not 

- [ ] Impossible to put proper spacing around code blocks => for this one I'll need more infos

Linked to #11510

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
Thanks for these quick fixes, the remaining stuff is fine. Honestly at some point in the future I think we should start looking at alternative wysiwyg as I think Draft.js is starting to show it's age.

File: packages/core/admin/admin/src/content-manager/components/Wysiwyg/Editor.js [hunk 1]
```diff
@@ -25,10 +25,12 @@ const Editor = ({
       lineWrapping: true,
       extraKeys: {
         Enter: 'newlineAndIndentContinueMarkdownList',
-        Tab: false,
-        'Shift-Tab': false,
+        // Leaving this commented for now
+        // Tab: false,
+        // 'Shift-Tab': false,
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/content-manager/components/Wysiwyg/Editor.js:30]: Do we still need that ?

File: packages/core/admin/admin/src/content-manager/components/Wysiwyg/EditorWrapper.js [hunk 2]
```diff
@@ -191,7 +179,7 @@ const EditorWrapper = styled.div`
   .CodeMirror-scrollbar-filler,
   .CodeMirror-gutter-filler {
     position: absolute;
-    /* z-index: 6; */
+    z-index: 1;
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/content-manager/components/Wysiwyg/EditorWrapper.js:182]: Does this work with our design system components z indices ?

File: packages/core/admin/admin/src/content-manager/components/Wysiwyg/utils/utils.js [hunk 3]
```diff
@@ -133,16 +133,48 @@ export const markdownHandler = (editor, markdownType) => {
...
+    const listToInsert = insertListOrTitle(listType);
+    const lineContent = editor.current.getLine(currentLine);
+
+    const textToInsert = listToInsert + lineContent;
+    editor.current.setSelection(
+      { line: currentLine, ch: 0 },
+      { line: currentLine, ch: lineContent.length }
+    );
+    editor.current.replaceSelection(textToInsert);
+  }
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/content-manager/components/Wysiwyg/utils/utils.js:176]: Seem to be a huge operation on the editor. Do we have some tests ?

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

File: packages/core/admin/admin/src/content-manager/components/Wysiwyg/EditorWrapper.js (hunk 2 above)
Comment (soupette) [reply packages/core/admin/admin/src/content-manager/components/Wysiwyg/EditorWrapper.js:182]: Yes it does, this is why I have set it to 1.

File: packages/core/admin/admin/src/content-manager/components/Wysiwyg/utils/utils.js (hunk 3 above)
Comment (soupette) [reply packages/core/admin/admin/src/content-manager/components/Wysiwyg/utils/utils.js:176]: The current test suite has some tests regarding the ul and ol lists and the addition does not break them.

**Review by HichamELBSI — APPROVED** [review]
LGTM

**Comment by soupette** [comment]
> Thanks for these quick fixes, the remaining stuff is fine. Honestly at some point in the future I think we should start looking at alternative wysiwyg as I think Draft.js is starting to show it's age.

We are using a brand new lib, Codemirror the same one that we use for the JSON input...

**Comment by derrickmehaffy** [comment]
> > Thanks for these quick fixes, the remaining stuff is fine. Honestly at some point in the future I think we should start looking at alternative wysiwyg as I think Draft.js is starting to show it's age.
> 
> We are using a brand new lib, Codemirror the same one that we use for the JSON input... This is why there's a lot of improvement why this wysiwyg and some newly bugs as well.

That makes more sense, I was slightly confused about it's new layout but just assumed it was some config changes. there will likely be some growing pains for a while I think but we will handle them as they come. :)

**Pull request merged**